README.md

```diff
@@ -103,25 +103,95 @@ In CI linting, formatting and also run. These checks can be seen in the [CI work
+
+```bash
+docker run -p 3000:80 [TAG]
+```
+
+## Build Instructions
+
+### dist build
+
+Packaging prepared dist into [`nginx`](https://hub.docker.com/_/nginx)([server configuration](./nginx/nginx.conf)) docker image involves building the application on docker host machine from source.
+
+As a prerequisite you need to perform build of `dist` directory and move its content for specific application to `dist-result` directory.
+
+This is an example script written in bash with possible implementation (using `--env=mainnet` flag to control `.env` file):
+
+```bash
+app=[YOUR APP]
+flags="--env=mainnet" # change to use .env file for any other environment
+yarn install
+if [ "${app}" = "trading" ]; then
+  yarn nx export trading $flags
+  DIST_LOCATION=dist/apps/trading/exported
+else
+  yarn nx build ${app} $flags
+  DIST_LOCATION=dist/apps/${app}
+fi
+cp -r $DIST_LOCATION dist-result
+```
 
 You can build any of the containers locally with the following command:
 
 ```bash
-docker build --dockerfile dockerfiles/Dockerfile.cra . --build-arg APP=[YOUR APP] --tag=[TAG]
+docker build --dockerfile dockerfiles/dist.Dockerfile . --tag=[TAG]
 ```
 
-In order to run a container:
+### node in docker build
+
+Using multistage dockerfile dist is compiled using [node](https://hub.docker.com/_/node) image and later packed to nginx as in [dist build](#dist-build) example.
 
 ```bash
-docker run -p 3000:80 [TAG]
+docker build --build-arg APP=[YOUR APP] --build-arg NODE_VERSION=$(cat .nvmrc) --build-arg ENV_NAME=mainnet -t [TAG] -f docker/node.Dockerfile .
+```
+
+### Computing ipfs-hash of the build
+
+At the moment this feature is important only for `trading` (console) releases.
+
+Each docker build finishes with hash calculation for dist directory. Resulting hash is added to file named as `/ipfs-hash`. Once docker image is produced you can run following commad to display ipfs-hash:
+
+```bash
+docker run --rm [TAG] cat /ipfs-hash
+```
+
+To recalculate hash again you need to run
+
+```bash
+docker run --rm [TAG] ipfs add -rw /usr/share/nginx/html
+```
+
+or to skip progress bar:
+
+```bash
+docker run --rm [TAG] ipfs add -rwQ /usr/share/nginx/html
+```
+
+**updating hash:** recompiling dist directory (even if there are no changed to source code) results in different hash computed by ipfs command.
+
+### Verifying ipfs-hash of existing current application version
+
+it is advised to compare computed hash with the one from github [release notes](https://github.com/vegaprotocol/frontend-monorepo/releases) to ensure you are running exact same version.
+
+1. run `docker pull vegaprotocol/<APP NAME>:<TAG>`
+2. follow instructions from [previous step](#computing-ipfs-hash-of-the-build) and use image name from step 1 as `[TAG]` to recalculate html directory
+3. you should see exactly same hash produced by ipfs command as one placed with the release notes
+4. if you want to extract dist from docker image to your local filesystem you can run following command:
```

Comment:
**thought (non-blocking)**: Again I think this could be a `makefile` target called something like `eject`, so that end to end it's more like

`make release-tag; make release-tag-ipfs; make eject`